src/main/java/hudson/remoting/Engine.java

```diff
@@ -1270,6 +1270,7 @@ private static SSLContext getSSLContext(List<X509Certificate> x509Certificates,
             SSLContext ctx = SSLContext.getInstance("TLS");
             // now we have our custom socket factory
             ctx.init(null, trustManagerFactory.getTrustManagers(), null);
+            sslContext = ctx;
```

Comment:
without this the value from `-cert` wasn't getting considered in websocket connection and it was failing with,
```
Mar 11, 2025 10:09:24 PM hudson.remoting.Launcher$CuiListener status
INFO: Failed to connect: Connection failed.
Mar 11, 2025 10:09:24 PM hudson.remoting.Launcher$CuiListener status
INFO: Waiting 1 second before retry
Mar 11, 2025 10:09:25 PM hudson.remoting.Launcher$CuiListener status
INFO: Failed to connect: Connection failed.
Mar 11, 2025 10:09:25 PM hudson.remoting.Launcher$CuiListener status
INFO: Waiting 3 seconds before retry
```

Reply:
Yeah looks like a regression in #690

> I did not test -cert because…

Could definitely benefit from a core IT using https://github.com/jenkinsci/jenkins-test-harness/pull/858!

Reply:
Ugh, sorry about that. I'm a bit surprised that nobody has reported this yet, since that regression was introduced a year and a half ago.

Reply:
Either not that many people run controllers with self-signed certificates, or (what seems more likely) they just use `-noCertificateCheck` and do not bother with `-cert` at all.